### PR TITLE
Consider faction ownerships in item stolen checks (bug #4293)

### PR DESCRIPTION
**Pull request by Capostrophic**
[Bug #4293.](https://bugs.openmw.org/issues/4293)
If you try to sell Saltrice from Mages Guild-owned sacks to Ajira, a "This Saltrice is mine!" messagebox will appear along with "your crime has been reported" message and Ajira will attack you. As will everyone else in the building.

Same happens in the original engine, although Ajira herself doesn't attack the player for some reason.

### CODE REVIEW

File: apps/openmw/mwmechanics/mechanicsmanagerimp.cpp [hunk 1]
```diff
@@ -992,14 +992,19 @@ namespace MWMechanics
...
-    bool MechanicsManager::isItemStolenFrom(const std::string &itemid, const std::string &ownerid)
+    bool MechanicsManager::isItemStolenFrom(const std::string &itemid, const MWWorld::Ptr& ptr)
     {
         StolenItemsMap::const_iterator it = mStolenItems.find(Misc::StringUtils::lowerCase(itemid));
         if (it == mStolenItems.end())
             return false;
+
         const OwnerMap& owners = it->second;
+        const std::string ownerid = ptr.getCellRef().getRefId();
+        const std::string factionid = ptr.getClass().getPrimaryFaction(ptr);
```
Comment (akortunov) [inline apps/openmw/mwmechanics/mechanicsmanagerimp.cpp]: What will happen if the trader has no faction? Just unnecessary search for empty string?

Comment (Capostrophic) [reply apps/openmw/mwmechanics/mechanicsmanagerimp.cpp]: Ouch. Thanks.

**Comment by Capostrophic** [comment]
Wait. I forgot to change confiscation function.

File: apps/openmw/mwmechanics/mechanicsmanagerimp.cpp [hunk 2]
```diff
@@ -1026,8 +1026,19 @@ namespace MWMechanics
...
         Owner owner;
-        owner.first = victim.getCellRef().getRefId();
-        owner.second = false;
+
+        if (item.getCellRef().getOwner() == victim.getCellRef().getRefId())
+        {
+            owner.first = victim.getCellRef().getRefId();
+            owner.second = false;
+        }
+        else if (!victim.getClass().getPrimaryFaction(victim).empty()) // Assume the item is faction-owned
```
Comment (akortunov) [inline apps/openmw/mwmechanics/mechanicsmanagerimp.cpp]: Maybe move `victim.getClass().getPrimaryFaction(victim)` to variable?

**Comment by zinnschlag** [comment]
For future reference: We are currently in the process of making a change to the PR procedure (just test-run, will be reviewed again after the 0.45 release). For any new PR (related to the issue), please add the issue description to the CHANGELOG.md file.

**Comment by Capostrophic** [comment]
Got it, though I hope 0.45.0 is not going to be a Christmas release.

**Comment by psi29a** [comment]
That would be 1.0 ;)

**Comment by testman42** [comment]
```css
>implying implications
```

**Comment by psi29a** [comment]
It was never said which Christmas. :P

**Pull request merged**